apps/client/src/common/components/client-modal/RedirectClientModal.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from 'react';
+import {
+  Button,
+  Input,
+  InputGroup,
+  InputLeftAddon,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+} from '@chakra-ui/react';
+import { ClientList } from 'ontime-types';
+
+import style from './ClientModal.module.scss';
+
+interface RedirectClientModalProps {
+  onClose: () => void;
+  isOpen: boolean;
+  id: string;
+  clients: ClientList;
+  onSubmit: (path: string) => void;
+}
+
+export function RedirectClientModal(props: RedirectClientModalProps) {
+  const { onClose, isOpen, id, clients, onSubmit } = props;
+  const [path, setPath] = useState('');
+
+  // when clientlist and id is populated get the relevant path to use a default value
+  useEffect(() => {
+    if (clients[id]?.path) setPath(clients[id].path);
+  }, [clients, id]);
```

Comment:
question: usually combinations of useState and useEffect are incorrect.

can we combine this by initialising the state?

```ts
const [path, setPath] = useState(clients[id]?.path || '');
```